### ISSUE TITLE
Question about default '\n\n' padding in TikTokenTokenizer

### ISSUE THREAD

**Issue by vvatter** [pytorch/torchtune#943]
Hi,

I'm currently using Torchtune (0.1.1)for fine-tuning llama-3 and have come across behavior in the `TikTokenTokenizer` that I find puzzling. By default, `tokenize_messages` calls `tokenize_message` with `tokenize-header=True`, which then causes the relevant message to be prepended with '\n\n'. Here's the relevant snippet from `tokenize_message`:
```
if tokenize_header:
    tokenized_header = (
        [self.start_header_id]
        + self.encode(message.role.strip(), add_bos=False, add_eos=False)
        + [self.end_header_id]
        + self.encode("\n\n", add_bos=False, add_eos=False)
    )
```

I've noticed that llama-3 does not prepend this '\n\n' to its responses by default, and so I'm wondering why the padding is added.

In my particular use-case, I would like llama-3 to select a single-token answer, and so it would be nice if I could train it on just those single tokens not prepended by '\n\n'. Is there any way to turn the padding off?

Thanks for any insight and thanks for the great project!

**Comment by ebsmothers** [comment]
Hi @vvatter thanks for creating the issue. To make sure I understand correctly, you still want to tokenize the header, but you do not want to include the tokenized newlines, is that right? For what it's worth we are closely following the implementation in the meta-llama repo, which adds the newlines [here](https://github.com/meta-llama/llama3/blob/main/llama/tokenizer.py#L211) for the instruction-tuned models ([ref](https://github.com/meta-llama/llama3/tree/main?tab=readme-ov-file#instruction-tuned-models)). But if you want to do without them (and keep the remainder of the header), one option would be to inherit from `TikTokenTokenizer` in your own code and override the `tokenize_message` method yourself to get the desired behavior. 

Let me know if this makes sense for what you're trying to do. If not, happy to discuss further to figure out what the right approach here is.

**Comment by vvatter** [comment]
@ebsmothers , thanks for explaining where those newlines come from. You did understand my intentions correctly, but your suggestion worked great for my use case, so thank you!

I'm still confused by the newlines in the following sense, though. If I ask llama3-instruct a multiple choice question, it seems like it always responds to me with something like
```
<|start_header_id|>assistant<|end_header_id|>C<|eot_id|>
```
and not with something like
```
<|start_header_id|>assistant<|end_header_id|>\n\nC<|eot_id|>
```

This is one reason why I wanted to train it on outputs without the \n\n prefix. (The other reason is that I want to be lazy and read its output just by checking the logits of the first token.) Anyway like I said, I'm happy now.

**Comment by kartikayk** [comment]
@vvatter I'm closing this issue, but please don't hesitate to reopen if there are follow ups for us!
